src/lucky/renderable.cr

```diff
@@ -69,29 +69,43 @@ module Lucky::Renderable
 
   # Render an HTMLPage with a status other than 200
   #
+  # The status can either be a Number, a HTTP::Status, or a Symbol that corresponds to the HTTP::Status.
+  #
   # ```
   # class SecretAgents::Index < BrowserAction
   #   get "/shhhh" do
   #     html_with_status IndexPage, 472, message: "This page can only be seen with special goggles"
   #   end
   # end
   # ```
+  # See Crystal's
+  # [HTTP::Status](https://crystal-lang.org/api/latest/HTTP/Status.html)
+  # enum for more available http status codes.
   macro html_with_status(page_class, status, **assigns)
-    {% if !status.is_a?(NumberLiteral) %}
+    {% if status.is_a?(NumberLiteral) %}
+      html {{ page_class }}, _with_status_code: {{ status }}, {{ **assigns }}
+    {% elsif status.is_a?(SymbolLiteral) %}
+      html {{ page_class }}, _with_status_code: HTTP::Status::{{ status.upcase.id }}.value, {{ **assigns }}
+    {% elsif status.resolve.is_a?(NumberLiteral) %}
```

Comment:
This is for the `HTTP::Status` enum case. When in "macro land" status is a `Crystal::Macros::Path`. At first i tried `status.is_a?(HTTP::Status)` however that would return false.

Reply:
Ah, interesting. So the `Crystal::Macros::Path#resolve` will turn that enum in to an Integer?

Reply:
That's right, so it will actually work with all Enums with the current implementation, maybe we should add a check to see if the `Path` starts with `HTTP::Status`

Reply:
That might not be a bad idea. You wouldn't want to accidentally send out a status of `4` :joy:  Then again.. maybe you do? :thinking:  The whole reason for this method was more of an escape hatch since for the most part, you'd want to just use the `html` macro...

Reply:
I added the check, if you really wanted to sent out a status of `4` you could just pass a `Number`. Here be dragons 🐉.
Maybe it won't be an uncommon escsape hatch for long, the `Turbo` library from `Hotwired` expects a `422 unprocessable entity` when a form fails with validation errors.

Reply:
Sweet. Sounds good. Thanks!